src/dotVariant.Generator/Diagnose.cs

```diff
@@ -240,7 +246,7 @@ private static class CheckThat
             {
                 foreach (var p in options)
                 {
-                    if (p.Type.TypeKind == TypeKind.Interface)
+                    if (p.Type.TypeKind == TypeKind.Interface && CheckNoImplicitConversionAttribute(p))
```

Comment:
Please add a new class to [NoImplicitConversionForInterfaces.cs](https://github.com/mknejp/dotvariant/blob/stable/src/dotVariant.Generator.Test/diagnostics/NoImplicitConversionForBaseClasses.cs) with the attribute present to test that the warning is silenced.